### DISCUSSION TITLE
gllvm for estimating latent factors like confirmatory factor analysis in structural equation models

### DISCUSSION THREAD

**Discussion by joh4nd** [JenniNiku/gllvm#128] [Q&A] [answered]
Is gllvm applicable to other fields? is a certain data matrix required? 

As far as I can tell, gllvm:
- enables estimating logit family link functions between latent variables and ordinal data.
- is designed with a particular field in mind, i.e. abundance and ordination. (I am unfamiliar with the domain concepts.)

Searches on the statistical terms led me to finding this R package. I want to use it for estimating traits on individuals, people based on surveys/questionaires. Other packages (lavaan, psych, OpenMx) for CFA and SEM do not have the capacity to compute link functions of any family. Instead, these packages are based on estimating polychoric correlations between ordinal variables, and feeding such matrix to SEM/CFA computing the latent factor loadings and what not.

- units/observations in rows and variables/items in columns
- variables take ordinal values on a 0-3 scale
- latent scores are estimated per unit/row
- latent loadings on each variable are computed on the sample (across rows)

**Comment by BertvanderVeen** [comment, marked as answer]
Thanks for your question!

Excellent point. The r-package was constructed with community ecology in mind, but the methods are definitely applicable to other fields. No large requirements are placed on the data matrix, other than those you described. The main difficulty I find with the analysis of survey data is the usual inclusion of an "I don't know answer" that doesn't really fit into the ordinal scale as it is (essentially) missing data, and I have yet to find an appropriate way to deal with that.

It might be that setting those observations to NA might just do the trick.

It is useful to note that gllvm allows application to a range of data types as described in the [vignettes](https://cran.r-project.org/web/packages/gllvm/vignettes/vignette1.html), and ordinal data is only one response type.

**Comment by joh4nd** [comment]
Thanks a lot for the feedback, Bert, and so quick :)

Great news, if I understand, that suggests a script like below for a model of the relationship between two latent variables and say six observed variables. But I would appreciate your feedback since I am not entirely sure though I looked at the documentation.

```
# how do you specify two latent factors that the observed items load on?
model <- 'F1 ~ item1 + item2 + item3
F2 ~ item4 + item5 + item6
'

# is it possible to use clusters and weights in the studyDesign argument?
sdesign <- survey2023[, c(cluster, pw)]

fit <- gllvm(data=survey2023[, c(item1, item2, item3, item4, item5, item6)], formula=model, studyDesign=sdesign, family="ordinal", num.lv=2, link="logit", zeta.struc="common")

summary(fit)
```


Btw,  you are entirely right about recoding "don't know" as missing values (and use either pairwise deletion or FIML). For factual questions, "don't know" simply means "don't know", and, as such, it can be recoded as a wrong answer.

And FYI, the packages I mentioned (and possibly also the Stata software and its GSEM command) has features of possible interest. E.g. goodness of fit indices like RMSEA, SRMR, TLI, and CFI (albeit e,g, Stata software does not provide many of these indices for models estimated with ordered logit family link functions. lavaan and psych does with polychoric matrices).

**Comment by BertvanderVeen** [comment]
- you can use use studyDesign to account for clustered rows but not weights (though depends a bit what you mean by weights) 
- all columns are by design connected to all latent variables,
- the model you've specified looks fine, though I'm not sure what the purpose is of c(item1,... item6). 
- Note that, though it is on the to do list, it is not currently possible to specify a logit-link function for the ordinal model in gllvm. The only available option at the moment is the probit link. This will probably be expanded in the near future

**Reply by joh4nd** [reply]
Nice, thanks. 

- Being able to load latent factors on different, specified columns would be major. I guess that would make it a SEM package.

- By weights I meant sampling weights or (inverse) probability weights. In survey lingo, there are a couple of different ones, so I understand that you asked.
